apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderTest.java

```diff
@@ -451,48 +530,327 @@ public void testFindByPatientWhenNoClaimsFoundExpectEmptyBundle() {
+    Root<CarrierClaim> carrierRoot = mock(Root.class);
+    setupClaimEntity(carrierEntityManager, ClaimTypeV2.CARRIER, carrierMockCriteria, carrierRoot);
+
+    // mock DMEClaim data
+    CriteriaQuery<DMEClaim> dmeMockCriteria = mock(CriteriaQuery.class);
+    Root<DMEClaim> dmeRoot = mock(Root.class);
+    setupClaimEntity(dmeEntityManager, ClaimTypeV2.DME, dmeMockCriteria, dmeRoot);
+
+    // mock SNFClaim data
+    CriteriaQuery<SNFClaim> snfMockCriteria = mock(CriteriaQuery.class);
+    Root<SNFClaim> snfRoot = mock(Root.class);
+    setupClaimEntity(snfEntityManager, ClaimTypeV2.SNF, snfMockCriteria, snfRoot);
+
+    PatientClaimsEobTaskTransformerV2 carrierTask =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+    carrierTask.setEntityManager(carrierEntityManager);
+    carrierTask.setupTaskParams(
+        carrierClaimTransformer,
+        ClaimTypeV2.CARRIER,
+        Long.parseLong(BENE_ID),
+        Optional.empty(),
+        Optional.empty(),
+        true);
+    carrierTask.setIncludeTaxNumbers(true);
+    carrierTask.call();
+    assertTrue(carrierTask.ranSuccessfully());
+    assertTrue(carrierTask.eobsRemovedBySamhsaFilter() < 1);
+
+    PatientClaimsEobTaskTransformerV2 dmeTask =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+    dmeTask.setEntityManager(dmeEntityManager);
+    dmeTask.setupTaskParams(
+        dmeClaimTransformer,
+        ClaimTypeV2.DME,
+        Long.parseLong(BENE_ID),
+        Optional.empty(),
+        Optional.empty(),
+        true);
+    dmeTask.setIncludeTaxNumbers(true);
+    dmeTask.call();
+    assertTrue(dmeTask.ranSuccessfully());
+    assertTrue(dmeTask.eobsRemovedBySamhsaFilter() < 1);
+
+    PatientClaimsEobTaskTransformerV2 snfTask =
+        new PatientClaimsEobTaskTransformerV2(
+            metricRegistry, mockSamhsaMatcher, mockDrugDisplayLookup, mockNpiOrgLookup);
+    snfTask.setEntityManager(snfEntityManager);
+    snfTask.setupTaskParams(
+        snfClaimTransformer,
+        ClaimTypeV2.SNF,
+        Long.parseLong(BENE_ID),
+        Optional.empty(),
+        Optional.empty(),
+        true);
+    snfTask.call();
+    assertTrue(snfTask.ranSuccessfully());
+    assertTrue(snfTask.eobsRemovedBySamhsaFilter() < 1);
   }
 
   /**
-   * Verifies that {@link R4ExplanationOfBenefitResourceProvider#findByPatient} with <code>
+   * Verifies that processing multiple {@link PatientClaimsEobTaskTransformerV2} tasks with <code>
    * excludeSAMHSA=false</code> does not call the filter for SAMHSA-related claims.
    */
   @Test
   public void testFindByPatientWhenExcludeSamshaFalseExpectNoFiltering() {
-    eobProvider.findByPatient(patientParam, null, null, "false", null, null, null, requestDetails);
-
-    verify(samhsaMatcher, never()).test(testEob);
```

Comment:
It's a shame to lose this `verify()` call.  It's a great way to ensure no calls were made to the matcher.  Is it no longer possible to do this?  Or to do a `verifyNoInterations(...);`?

Reply:
I think it's because the samhsa matching was moved into the task class, so the matcher isn't tested here anymore.